client/me/social-login/index.jsx

```diff
@@ -48,11 +53,17 @@ class SocialLogin extends Component {
 		debug( this.constructor.displayName + ' React component is unmounting.' );
 	}
 
+	refreshUser() {
+		user.fetch();
+		this.setState( { fetchingUser: true } );
```

Comment:
Should we check `if ( ! this.state.fetching ) { ... }` ?

Reply:
I don't think it's necessary. If we're fetching, then the button is disabled and we can't trigger another refresh anyway.
There's however the case where the fetch request fails which is not handled here but that would mean changing some code un lib/user and I'm already investigating this in https://github.com/Automattic/wp-calypso/pull/17772